drivers/bluetooth/hci/CMakeLists.txt

```diff
@@ -29,9 +37,33 @@ zephyr_library_sources_ifdef(CONFIG_BT_CYW208XX  hci_ifx_cyw208xx.c)
 
 zephyr_library_sources_ifdef(CONFIG_BT_STM32_IPM   ipm_stm32wb.c)
 zephyr_library_sources_ifdef(CONFIG_BT_STM32WBA    hci_stm32wba.c)
+if(CONFIG_DT_HAS_ST_HCI_STM32WBA_ENABLED AND NOT CONFIG_ZEPHYR_HAL_STM32_MODULE_BLOBS)
+  message(FATAL_ERROR "
+    ------------------------------------------------------------------------
+    Missing blobs for STM32WBA HCI driver, please install by running:
+        $ west update
+        $ west blobs fetch hal_stm32
+    ------------------------------------------------------------------------")
+endif()
```

Comment:
You could use the `zephyr_verify_blobs` function here to verify the `sha256` too.

```cmake
if(CONFIG_DT_HAS_ST_HCI_STM32WBA_ENABLED)
  zephyr_blobs_verify(MODULE hal_stm32 REQUIRED)
endif()
```

Reply:
> You could use the `zephyr_verify_blobs` function here to verify the `sha256` too.
> 
> ```cmake
> if(CONFIG_DT_HAS_ST_HCI_STM32WBA_ENABLED)
>   zephyr_blobs_verify(MODULE hal_stm32 REQUIRED)
> endif()
> ```

@pdgendt thanks, that looks like a nicer way to do it. It also helped me identify one more HCI driver with a blobs dependency: BT_H4_NXP_CTRL. The hal_nxp check is currently done in `modules/hal_nxp/bt_controller/CMakeLists.txt`:
https://github.com/zephyrproject-rtos/zephyr/blob/f29377a12cb5dd2ee845f21b8319c0c4ee2cca40/modules/hal_nxp/bt_controller/CMakeLists.txt#L19

I think it would be good that the location for these checks would be consistent for all HCI drivers. The check done for NXP also seems more specific, i.e. it's not checking for all blobs, rather just the ones needed for Bluetooth. Maybe the same should be done for all HCI drivers, i.e. the checks should move somewhere under `modules/*/ rather than `drivers/bluetooth/hci/Kconfig`?

Reply:
@lylezhu2012 @dleach02 any thoughts on the above?

Reply:
> You could use the zephyr_verify_blobs function here to verify the sha256 too.

@pdgendt  thanks for remembering that function :slightly_smiling_face:

Reply:
> @lylezhu2012 @dleach02 any thoughts on the above?

Yes. Actually, not all blobs may be used in this compilation. Checking specific blobs can optimize compilation speed.